lib/Twig/Loader/Filesystem.php

```diff
@@ -89,7 +89,9 @@ public function addPath($path, $namespace = self::MAIN_NAMESPACE)
     {
         // invalidate the cache
         $this->cache = $this->errorCache = array();
```

Comment:
now the cache is cleared even if the method doesn't do anything. so you can see that the fix doesn't make sense in twig.